internal/parse/parse.go

```diff
@@ -37,6 +37,56 @@ var (
 	errBadVolSrc    = errors.New("must set volume source")
 )
 
+// GetBuildOutput is responsible for parsing custom build output argument.
+// Takes `buildOutput` as string and returns path where output has to be generated.
+func GetBuildOutput(buildOutput string) (string, bool, bool, error) {
+	isDir := true
+	isStdout := false
+	typeSelected := false
+	pathSelected := false
+	path := ""
+	if len(buildOutput) == 1 && buildOutput == "-" {
+		// Feature parity with buildkit, output tar to stdout
+		// Read more here: https://docs.docker.com/engine/reference/commandline/build/#custom-build-outputs
+		return "", false, true, nil
+	}
+	if strings.Contains(buildOutput, ",") {
+		tokens := strings.SplitN(buildOutput, ",", 2)
+		if len(tokens) != 2 {
+			return path, isDir, isStdout, fmt.Errorf("invalid build output options %q", buildOutput)
+		}
+		for _, option := range tokens {
+			arr := strings.SplitN(option, "=", 2)
+			if len(arr) != 2 {
+				return path, isDir, isStdout, fmt.Errorf("invalid build output options %q", buildOutput)
+			}
+			if arr[0] == "type" && !typeSelected {
+				typeSelected = true
+				if arr[1] == "local" {
+					isDir = true
+				} else if arr[1] == "tar" {
+					isDir = false
+				} else {
+					return path, isDir, isStdout, fmt.Errorf("invalid type %q selected for build output options %q", arr[1], buildOutput)
+				}
+			}
+			if arr[0] == "dest" && !pathSelected {
+				pathSelected = true
```

Comment:
It would be helpful for this to loudly complain about duplicate type/dest keys, instead of silently ignoring data.

Reply:
yes added a check for duplicates.